spec/helpers/bookmarks_helper_spec.rb

```diff
@@ -0,0 +1,223 @@
+require "spec_helper"
+
+describe BookmarksHelper do
+  let(:bookmarker) { create(:user) }
+  let(:external_work_bookmark) { create(:external_work_bookmark, pseud: bookmarker.default_pseud) }
+  let(:series_bookmark) { create(:series_bookmark, pseud: bookmarker.default_pseud) }
+  let(:work_bookmark) { create(:bookmark, pseud: bookmarker.default_pseud) }
+  let(:external_work) { external_work_bookmark.bookmarkable }
+  let(:series) { series_bookmark.bookmarkable }
+  let(:series_creator) { series.users.first }
+  let(:work) { work_bookmark.bookmarkable }
+  let(:work_creator) { work.users.first }
+
+  describe "#css_classes_for_bookmark_blurb" do
+    let(:default_classes) { "bookmark blurb group" }
+
+    context "when bookmarkable is ExternalWork" do
+      it "returns string with default classes, creation info, and bookmarker info" do
+        result = helper.css_classes_for_bookmark_blurb(external_work_bookmark)
+        expect(result).to eq("#{default_classes} external-work-#{external_work.id} user-#{bookmarker.id}")
+      end
+
+      context "when bookmark is updated" do
+        it "returns same string with different cache key" do
+          original_cache_key = "#{external_work.cache_key_with_version}_#{external_work_bookmark.cache_key}/blurb_css_classes"
+          expected_classes = "#{default_classes} external-work-#{external_work.id} user-#{bookmarker.id}"
+          expect(helper.css_classes_for_bookmark_blurb(external_work_bookmark)).to eq(expected_classes)
+
+          travel(1.day)
+          external_work_bookmark.update(bookmarker_notes: "New note")
+          expect(helper.css_classes_for_bookmark_blurb(external_work_bookmark.reload)).to eq(expected_classes)
+          expect(original_cache_key).not_to eq("#{external_work.cache_key_with_version}_#{external_work_bookmark.cache_key}/blurb_css_classes")
+          travel_back
+        end
+      end
+    end
+
+    context "when bookmarkable is Series" do
+      it "returns string with default classes, creation and creator info, and bookmarker info" do
+        result = helper.css_classes_for_bookmark_blurb(series_bookmark)
+        expect(result).to eq("#{default_classes} series-#{series.id} user-#{series_creator.id} user-#{bookmarker.id}")
+      end
+
+      context "when bookmarker is also series creator" do
+        before do
+          series.creatorships.first.update(pseud_id: bookmarker.default_pseud_id)
+        end
+
+        it "only includes the user id once" do
+          result = helper.css_classes_for_bookmark_blurb(series_bookmark)
+          expect(result).to eq("#{default_classes} series-#{series.id} user-#{bookmarker.id}")
+        end
+      end
+
+      context "when series is updated" do
+        context "when new creator is added" do
```

Comment:
I only tested one scenario for updating series and works, but I can add more (e.g., removing creators and making the work unrevealed or anonymous).

Reply:
Let's not. If those scenarios fail to change the cache keys, I expect the `css_classes_for_creation_blurb` specs in spec/helpers/application_helper_spec.rb to fail first.